docs/user-guide/bandwidth-resources.md

```diff
@@ -0,0 +1,171 @@
+containers scheduled to the node is less than the capacity of the node.  Note
+that although actual bandwidth resource usage on nodes is very low, the
+scheduler will still refuse to place pods onto nodes if the capacity check
+fails.  This protects against a resource shortage on a node when resource usage
+later increases, such as due to a daily peak in request rate.
+
+Note: Although the scheduler normally spreads pods out across nodes, there are currently some cases
+where pods with no limits (unset values) might all land on the same node.
+
+## How Pods with Resource Limits are Run
+
+When kubelet starts a container of a pod, it manages the network resource consumption based
+on the pod's IP address, using the `tc` command.
+
+- The `spec.container[].resources.limits.ingress` or `egress` is summed for all containers in the pod.
+- The `tc` command is used to create a bandwidth limit classes for ingress and egress. Different classes are used for each..
+- The `tc` command is used to apply this bandwidth limit class to all network traffic either originating (egress) or arriving (ingress) to the Pod's IP address.
+
+`tc` is configured to use the hierarchical token bucket (`htb`) algorithm for bandwidth shaping.  Packets are slowed down in the kernel's queues in
+order to achieve the desired bandwidth restrictions.
+
+## Monitoring Bandwidth Resource Usage
+
+It is not currently possible to monitor the bandwidth usage of a pod.  Support for this is planned.
+
+## Troubleshooting
+
+### My pods are pending with event message failedScheduling
+
+If the scheduler cannot find any node where a pod can fit, then the pod will remain unscheduled
+until a place can be found.    An event will be produced each time the scheduler fails to find a
+place for the pod, like this:
+
+```console
+$ kubectl describe pods/frontend | grep -A 3 Events
+Events:
+  FirstSeen				LastSeen			Count	From SubobjectPath	Reason			Message
+  Tue, 30 Jun 2015 09:01:41 -0700	Tue, 30 Jun 2015 09:39:27 -0700	128	{scheduler }            failedScheduling	Error scheduling: For each of these fitness predicates, pod frontend failed on at least one node: PodFitsResources.
+```
+
+If a pod or pods are pending with this message, then there are several things to try:
+- Add more nodes to the cluster.
+- Terminate unneeded pods to make room for pending pods.
+- Check that the pod is not larger than all the nodes.  For example, if all the nodes
+have a capacity of `ingress: 10Mi`, then a pod with a limit of `ingress: 11Mi` will never be scheduled.
+
+You can check node capacities with the `kubectl get nodes -o <format>` command.
+Here are some example command lines that extract just the necessary information:
+- `kubectl get nodes -o yaml | grep '\sname\|cpu\|memory'`
+- `kubectl get nodes -o json | jq '.items[] | {name: .metadata.name, cap: .status.capacity}'`
+
+The [resource quota](../admin/resource-quota.md) feature can be configured
```

Comment:
cc @derekwaynecarr re. whether the quota or limitrange plugins need any updates to handle these resources.